sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -2515,4 +2515,35 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
     checkCount(2)
     countForMalformedJSON(0, Seq(""))
   }
+
+  test("saving pretty JSON in the mutliLine mode") {
+    withTempPath { path =>
+      val df = spark.range(1).select(
```

Comment:
Hm .. wait .. does this work for multiple records?

Reply:
let's just allow this option in JSON function only for now if not (I guess it's not from a cursory look).